1Homework18.02.22/BWT/BWT/Program.cs

```diff
@@ -0,0 +1,313 @@
+        {
+            var line = "ABACABAAAA";
+            var array1 = new int[10] {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+            var array2 = new int[10] {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+            var array1Answer = new int[10] {0, 2, 4, 6, 7, 8, 9, 1, 5, 3};
+            var array2Answer = new int[10] {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+            InsertionSort(array1, line, 0, 0, 9);
+            InsertionSort(array2, line, 2, 4, 9);
+            return AreArraysEquals(array1, array1Answer) && AreArraysEquals(array2, array2Answer);
+        }
+        private static bool TestBWT()
+        {
+            var string1 = "ABACABA";
+            var string1Answer = "BCABAAA";
+            int index1 = 0;
+            int index1Answer = 2;
+            var string2 = "mississippi";
+            var string2Answer = "pssmipissii";
+            int index2 = 0;
+            int index2Answer = 4;
+            var string3 = "";
+            var string3Answer = "";
+            int index3 = 0;
+            (string1, index1) = BWT(string1);
+            (string2, index2) = BWT(string2);
+            (string3, index3) = BWT(string3);
+            return String.Compare(string1, string1Answer) == 0
+                && String.Compare(string2, string2Answer) == 0
+                && String.Compare(string3, string3Answer) == 0
+                && index1 == index1Answer && index2 == index2Answer;
+        }
+
+        private static bool TestInverseBWT()
+        {
+            var string1Answer = "ABACABA";
+            var string1 = "BCABAAA";
+            int index1 = 2;
+            var string2Answer = "mississippi";
+            var string2 = "pssmipissii";
+            int index2 = 4;
+            var string3 = "";
+            var string3Answer = "";
+            int index3 = 0;
+            string1 = InverseBWT(string1, index1);
+            string2 = InverseBWT(string2, index2);
+            string3 = InverseBWT(string3, index3);
+            return String.Compare(string1, string1Answer) == 0
+                   && String.Compare(string2, string2Answer) == 0
+                   && String.Compare(string3, string3Answer) == 0;
+        }
+        
+        public static int Main(string[] args)
+        {
+            if (!TestHeapSort() || !TestInsertionSort() || !TestBWT() || !TestInverseBWT())
+            {
+                Console.WriteLine("Tests failed");
+                return -1;
+            }
+            Console.WriteLine("Enter a string without spaces:");
+            var input = Console.ReadLine();
+            if (input == null)
+            {
+                Console.WriteLine("Input error");
+                return -1;
+            }
+            int key = 0;
+            (input, key) = BWT(input);
+            if (key == -1)
+            {
+                Console.WriteLine("BTW failed");
+                return -1;
+            }
+            Console.WriteLine("The string after BWT: {0}", input);
```

Comment:
В современном C# принято пользоваться интерполяцией строк: 
```suggestion
            Console.WriteLine($"The string after BWT: {input}");
```